user/id.go

```diff
@@ -0,0 +1,101 @@
+package user
+
+import (
+	"fmt"
+
+	"github.com/mr-tron/base58"
+	"github.com/nspcc-dev/neofs-api-go/v2/refs"
+)
+
+// ID identifies users of the NeoFS system.
+//
+// ID is mutually compatible with github.com/nspcc-dev/neofs-api-go/v2/refs.OwnerID
+// message. See ReadFromV2 / WriteToV2 methods.
+//
+// Instances can be created using built-in var declaration.
+//
+// Note that direct typecast is not safe and may result in loss of compatibility:
+// 	_ = ID([25]byte{}) // not recommended
+type ID [25]byte
+
+// ReadFromV2 reads ID from the refs.OwnerID message. Returns an error if
+// the message is malformed according to the NeoFS API V2 protocol.
+//
+// See also WriteToV2.
+func (x *ID) ReadFromV2(m refs.OwnerID) error {
+	return x.Decode(m.GetValue())
+}
+
+// WriteToV2 writes ID to the refs.OwnerID message.
+// The message must not be nil.
+//
+// See also ReadFromV2.
+func (x ID) WriteToV2(m *refs.OwnerID) {
+	m.SetValue(x[:])
+}
+
+// Encode encodes ID into 25 bytes of dst.
+//
+// Zero ID is all zeros.
+//
+// Encode panics if dst length is less than 25.
+//
+// See also Decode.
+func (x ID) Encode(dst []byte) {
+	copy(dst, x[:])
+}
+
+// Decode decodes src bytes into ID.
+//
+// Decode expects that src has 25 bytes length. If the input is malformed,
+// Decode returns an error. In this case src remains unchanged.
+//
+// See also Encode.
+func (x *ID) Decode(src []byte) error {
+	if len(src) != 25 {
+		return fmt.Errorf("invalid length %d", len(src))
+	}
+
+	copy(x[:], src)
+
+	return nil
+}
+
+// EncodeToString encodes ID into NeoFS API protocol string.
+//
+// See also DecodeString.
+func (x ID) EncodeToString() string {
+	return base58.Encode(x[:])
+}
+
+// DecodeString decodes string into ID according to NeoFS API protocol. Returns
+// an error if s is malformed.
+//
+// Zero ID is base58 encoding of 25 zeros.
```

Comment:
shouldn't that line be in the `EncodeToString` method?

Reply:
Yes, but we've decided to invalidate zero ID instances, so this statement is going to be removed.